File: Codes/Hash/Non-Cryptographic/BKDRHash/BKDRHash.rb
# Hash function by Brian Kernighan & Dennis Ritchie
# Archive of Reversing.ID
# Non-Cryptographic Hash

def bkdrhash( key, len=key.length )
    seed  = 131    # 31 131 1313 13131 131313 etc..
    state = 0

    len.times{ |i|
        state = ( state * seed ) + key[i]
    }
    return state
end

File: Codes/Hash/Non-Cryptographic/BKDRHash/BKDRHash.java
/*
	Hash function by Brian Kernighan & Dennis Ritchie
    Archive of Reversing.ID
    Non-Cryptographic Hash
*/

// Fungsi ini merupakan method dari sebuah class
public long BKDRHash(String key)
{
    long seed = 131;    // 31 131 1313 13131 131313 etc..
    long state = 0;

    for(int i = 0; i < key.length(); i++)
    {
        state = (state * seed) + key.charAt(i);
    }

    return state;
}